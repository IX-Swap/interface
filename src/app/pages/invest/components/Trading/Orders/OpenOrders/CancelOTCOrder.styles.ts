import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  button: {
    [theme.breakpoints.down('lg')]: {
      background: '#EDE7FF',
      padding: theme.spacing(0.25, 0.5)
    }
  }
}))
