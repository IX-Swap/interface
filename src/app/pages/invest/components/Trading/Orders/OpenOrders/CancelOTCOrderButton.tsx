import { Box, Button, ButtonProps, CircularProgress } from '@mui/material'
import { useStyles } from 'app/pages/invest/components/Trading/Orders/OpenOrders/CancelOTCOrder.styles'
import { useCancelOTCOrder } from 'app/pages/invest/hooks/useCancelOTCOrder'
import React from 'react'
import { OTCOrder } from 'types/otcOrder'
export interface CancelOTCOrderButtonProps extends ButtonProps {
  order: OTCOrder
}

export const CancelOTCOrderButton = ({
  order,
  ...rest
}: CancelOTCOrderButtonProps) => {
  const classes = useStyles()
  const [cancelOrder, { status }] = useCancelOTCOrder(
    order._id,
    order.orderType
  )
  const handleClick = async () => {
    void cancelOrder()
  }
  return (
    <Box
      display='flex'
      justifyContent='center'
      alignItems={'center'}
      className={classes.button}
    >
      {status === 'loading' && <CircularProgress size={14} />}
      {status !== 'loading' && (
        <Button
          onClick={handleClick}
          variant='text'
          color='primary'
          size='small'
          {...rest}
        >
          Cancel
        </Button>
      )}
    </Box>
  )
}
