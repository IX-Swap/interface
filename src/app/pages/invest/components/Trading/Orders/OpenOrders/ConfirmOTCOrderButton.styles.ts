import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  loader: {
    width: theme.spacing(14),
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    [theme.breakpoints.down('md')]: {
      width: theme.spacing(10)
    }
  },
  buttonMobile: {
    textTransform: 'none',
    backgroundColor: theme.palette.background.paper,
    border: 'none',
    fontWeight: 400
  }
}))
