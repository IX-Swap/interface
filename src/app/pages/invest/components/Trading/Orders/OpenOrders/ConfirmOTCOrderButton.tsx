import { Box, Button, ButtonProps, CircularProgress } from '@mui/material'
import { useConfirmMyOrder } from 'app/pages/invest/hooks/useConfirmMyOrder'
import { usePairTokenAddressNetwork } from 'app/pages/invest/hooks/usePairTokenAddressNetwork'
import { useSendToken } from 'app/pages/invest/hooks/useSendToken'
import React, { useState } from 'react'
import { OTCOrder } from 'types/otcOrder'
export interface ConfirmOTCOrderButtonProps extends ButtonProps {
  order: OTCOrder
}
export const ConfirmOTCOrderButton = ({
  order,
  ...rest
}: ConfirmOTCOrderButtonProps) => {
  const { chainId, address } = usePairTokenAddressNetwork()
  const [loadingTransaction, setLoadingTransaction] = useState(false)
  const sendToken = useSendToken({
    address: address,
    tokenChainId: chainId
  })
  const [confirmMatch, { isLoading }] = useConfirmMyOrder()
  const handleClick = async () => {
    setLoadingTransaction(true)
    try {
      const sendingResult = await sendToken(
        order.matches?.matchedAmount ?? 0,
        order.matches?.ethAddress
      )
      if (sendingResult) {
        await confirmMatch({
          orderId: order._id,

          matchedOrderId: order.matches?.order ?? ''
        })
      }
    } catch {
      console.error('error confirming')
    } finally {
      setLoadingTransaction(false)
    }
  }

  return (
    <Box display='flex' justifyContent='center' alignItems={'center'}>
      {(isLoading || loadingTransaction) && <CircularProgress size={14} />}
      {!(isLoading || loadingTransaction) && (
        <Button
          disabled={isLoading || loadingTransaction}
          onClick={handleClick}
          variant='text'
          color='primary'
          size='small'
          {...rest}
        >
          Confirm
        </Button>
      )}
    </Box>
  )
}
