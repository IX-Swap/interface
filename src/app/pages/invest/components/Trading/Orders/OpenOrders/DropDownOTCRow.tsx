import { Box, IconButton } from '@mui/material'
import { OpenOrdersContext } from 'app/pages/invest/components/Trading/context/OpenOrdersContextWrapper'
import React, { useContext } from 'react'
import { OpenOTCOrder } from 'types/otcOrder'
import { Icon } from 'ui/Icons/Icon'
export interface DropDownOTCRowProps {
  order: OpenOTCOrder
}

export const DropDownOTCRow = ({ order }: DropDownOTCRowProps) => {
  const context = useContext(OpenOrdersContext)
  const isOpen = context?.isIndexOpen?.(order._id)
  const handleClick = () => {
    context?.toggleRow(order._id)
  }
  const iconName = isOpen === true ? 'chevron-up' : 'chevron-down'
  return (
    <Box display='flex' justifyContent='center' alignItems={'center'}>
      <IconButton data-testid='more-button' size='small' onClick={handleClick}>
        <Icon name={iconName} />
      </IconButton>
    </Box>
  )
}
