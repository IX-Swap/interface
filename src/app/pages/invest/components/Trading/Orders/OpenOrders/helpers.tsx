import React from 'react'
import { useMetamaskConnectionManager } from 'app/pages/invest/hooks/useMetamaskConnectionManager'
import { AccountState } from 'app/pages/invest/hooks/useMetamaskWalletState'
import { TableViewRendererProps } from 'components/TableWithPagination/TableView'
import { isNonEmptyArray } from 'helpers/arrays'
import { getRoundedPercentage } from 'helpers/numbers'
import { capitalizeFirstLetter } from 'helpers/strings'
import { useAppTheme } from 'hooks/useAppTheme'
import {
  ColumnOTCMatch,
  OpenOTCOrder,
  OTCMatch,
  OTCOrderStatus
} from 'types/otcOrder'
import StatusBar from './OTCOrderStatus'

export const needsConfirmation = (item: OpenOTCOrder) => {
  return item.orderType === 'SELL'
}

export const getColumnMatchedOrder = (
  row: OpenOTCOrder,
  matched: OTCMatch
): ColumnOTCMatch => {
  return {
    ...matched,
    pair: row.pair,
    createdAt: row.createdAt,
    orderType: row.orderType,
    parentOrder: row._id,
    parentAmount: row.amount
  }
}

export const useOpenOrderState = (
  props: TableViewRendererProps<OpenOTCOrder>
) => {
  const { columns, items, hasActions, loading } = props
  const { theme } = useAppTheme()
  const mobileRowColor = (item: OpenOTCOrder) => {
    if (!needsConfirmation(item)) {
      return 'initial'
    }
    return theme.palette.mode === 'light' ? '#F6F4FD' : '#494166'
  }
  const rowColor = (item: OpenOTCOrder) => {
    return theme.palette.mode === 'light' ? '#F6F4FD' : '#494166'
  }
  const columnCount = columns.length + Number(hasActions)
  const { accountState, isWhitelisted } = useMetamaskConnectionManager()
  const { found } = isWhitelisted
  const showEmptyState =
    accountState !== AccountState.SAME_CHAIN ||
    !found ||
    (items?.length === 0 && loading !== true)
  return { showEmptyState, columnCount, mobileRowColor, rowColor }
}

export const sortOpenOrders = (first: OpenOTCOrder, _: OpenOTCOrder) =>
  first?.matches !== undefined &&
  first.matches.length > 0 &&
  first.orderType === 'SELL'
    ? -1
    : 1

export const hasMatches = (item: OpenOTCOrder) =>
  item.matches !== undefined && item.matches.length > 0

export const hasApprovedMatches = (row: OpenOTCOrder) => {
  const approvedMatches =
    row?.matches?.filter(match => match.status === OTCOrderStatus.CONFIRMED) ??
    []
  return approvedMatches.length > 0
}

export const renderOpenOrderPercentage = (row: OpenOTCOrder) => {
  if (row.orderType === 'SELL') {
    return getRoundedPercentage({
      amount: +row.amount,
      matchedAmount: +row.amount - +row.availableAmount ?? 0
    })
  }
  const statusesWhereToShowFilled = [
    OTCOrderStatus.PENDING,
    OTCOrderStatus.COMPLETED,
    OTCOrderStatus.SETTLED
  ]
  const matchedAmounts = row?.matches
    ?.filter(match => statusesWhereToShowFilled.includes(match.status))
    .map(matched => matched.matchedAmount)
  const matchedAmount =
    matchedAmounts?.reduce(
      (previous, currentAmount) => +previous + +currentAmount,
      0
    ) ?? 0
  return matchedAmount === 0
    ? '0%'
    : getRoundedPercentage({
        amount: +row.amount,
        matchedAmount: matchedAmount
      })
}

export const renderOpenOrderStatus = (row: OpenOTCOrder) => {
  const percentageFilled = renderOpenOrderPercentage(row)

  return <StatusBar percentageFilled={percentageFilled} />
}

export const showCancelButton = ({ item }: { item: OpenOTCOrder }) => {
  if (item.orderType === 'SELL') {
    const disableCancelStatuses = [
      OTCOrderStatus.COMPLETED,
      OTCOrderStatus.CANCELLED,
      OTCOrderStatus.PENDING,
      OTCOrderStatus.SETTLED
    ]
    return !disableCancelStatuses.includes(item?.status)
  }
  const pendingMatches = item?.matches?.filter(
    item => item.status === OTCOrderStatus.PENDING
  )
  return !isNonEmptyArray(pendingMatches)
}

export const renderOTCOrderStatus = ({ item }: { item: OpenOTCOrder }) => {
  if (item.orderType === 'SELL') {
    return capitalizeFirstLetter(item?.status ?? '')
  }
  return 'Pending'
}
