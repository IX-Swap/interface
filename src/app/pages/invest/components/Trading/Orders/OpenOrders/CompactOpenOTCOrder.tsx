import { Box, Grid, TableBody, TableCell, TableRow } from '@mui/material'
import { useStyles } from 'app/pages/invest/components/Trading/Orders/OpenOrders/OpenOrders.styles'
import { CompactRowProps } from 'components/TableWithPagination/CompactRow'
import { TableViewRendererProps } from 'components/TableWithPagination/TableView'
import { getExpiresOrderMessage } from 'helpers/dates'
import get from 'lodash/get'
import React from 'react'
import { OTCOrder } from 'types/otcOrder'
import { needsConfirmation, sortOpenOrders, useOpenOrderState } from './helpers'
import { OpenOrdersEmptyState } from './OpenOrdersEmptyState'
import { OTCOrderActionsMobile } from './OTCOrderActionsMobile'

export interface CompactBodyProps<T> extends TableViewRendererProps<T> {
  renderRow?: (props: CompactRowProps<T>) => JSX.Element
}

export const CompactOpenOTCOrder = (props: CompactBodyProps<OTCOrder>) => {
  const { columns, items } = props
  const { showEmptyState, mobileRowColor } = useOpenOrderState(props)
  const classes = useStyles()

  if (showEmptyState) {
    return <OpenOrdersEmptyState />
  }
  const sorted = items?.sort(sortOpenOrders) ?? []
  return (
    <TableBody>
      {sorted.map((item, i) => (
        <TableRow
          key={i}
          style={{
            backgroundColor: mobileRowColor(item)
          }}
        >
          <TableCell>
            <Grid container spacing={1} alignItems='flex-start'>
              {columns.map(({ label, key, render }, index) =>
                index === 0 ? (
                  <React.Fragment key={key}>
                    <Grid item container justifyContent='space-between'>
                      <Grid item>
                        {key.length > 0 &&
                          (typeof render === 'function'
                            ? render(get(item, key), item)
                            : get(item, key))}
                      </Grid>
                      <Grid item>
                        <OTCOrderActionsMobile item={item} type='Cancel' />
                      </Grid>
                    </Grid>
                  </React.Fragment>
                ) : (
                  <React.Fragment key={key}>
                    <Grid item xs={6}>
                      {label}
                    </Grid>
                    <Grid
                      item
                      xs={6}
                      style={{ textAlign: 'right', fontWeight: 400 }}
                    >
                      {key.length > 0 &&
                        (typeof render === 'function'
                          ? render(get(item, key), item)
                          : get(item, key))}
                    </Grid>
                  </React.Fragment>
                )
              )}
              {needsConfirmation(item) && (
                <Grid
                  item
                  xs={12}
                  key={`${item._id}-timeout`}
                  style={{
                    backgroundColor: mobileRowColor(item),
                    textAlign: 'center'
                  }}
                >
                  <Box className={classes.infoCell}>
                    <Box className={classes.separator} />
                    {getExpiresOrderMessage(new Date(item.createdAt))}
                  </Box>
                </Grid>
              )}
              <Grid item xs={12}>
                <OTCOrderActionsMobile item={item} type='Confirm' />
              </Grid>
            </Grid>
          </TableCell>
        </TableRow>
      ))}
    </TableBody>
  )
}
