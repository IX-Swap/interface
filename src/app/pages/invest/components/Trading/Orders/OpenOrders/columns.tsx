import { formatDateToMMDDYY } from 'helpers/dates'
import {
  formatMoney,
  getFilledMatchesPercentage,
  getFilledPercentage,
  getOrderCurrency,
  renderMoney,
  renderTotal
} from 'helpers/numbers'
import { capitalizeFirstLetter } from 'helpers/strings'
import { renderTicker } from 'helpers/tables'
import { OTCOrder } from 'types/otcOrder'
import { TableColumn } from 'types/util'

export const columns: Array<TableColumn<OTCOrder>> = [
  {
    key: 'createdAt',
    label: 'Date',
    render: formatDateToMMDDYY
  },
  {
    label: 'Pair',
    key: 'pair.name',
    render: renderTicker
  },
  {
    key: 'orderType',
    label: 'Side',
    render: value => capitalizeFirstLetter(value)
  },
  {
    key: 'price',
    label: 'Price',
    render: (value, row) => formatMoney(value, getOrderCurrency(row), false)
  },
  {
    key: 'availableAmount',
    label: 'Matched amount',
    align: 'center',
    render: (value, row) =>
      formatMoney(row?.matches?.matchedAmount ?? 0, '', false)
  },
  {
    key: 'amount',
    label: 'Total',
    render: (_, row) =>
      renderTotal({ amount: row.amount, price: row.price, row })
  },
  {
    key: '_id',
    label: 'Filled',
    render: (_, row) =>
      getFilledMatchesPercentage({
        amount: row.amount,
        matchedAmount: row.matches?.matchedAmount ?? 0
      })
  }
]

export const compactColumns: Array<TableColumn<OTCOrder>> = [
  {
    label: 'Pair',
    key: 'pair.name',
    render: renderTicker
  },
  {
    key: 'amount',
    label: 'Amount',
    render: renderMoney
  },
  {
    key: 'orderType',
    label: 'Side',
    render: value => capitalizeFirstLetter(value)
  },
  {
    key: 'price',
    label: 'Price',
    render: (value, row) => formatMoney(value, getOrderCurrency(row), false)
  },
  {
    key: 'availableAmount',
    label: 'Total',
    render: (_, row) =>
      renderTotal({ amount: row.amount, price: row.price, row })
  },
  {
    key: '_id',
    label: 'Filled',
    render: (_, row) =>
      getFilledPercentage({
        amount: row.amount,
        availableAmount: row.availableAmount
      })
  },
  {
    key: 'createdAt',
    label: 'Date',
    render: formatDateToMMDDYY
  }
]
