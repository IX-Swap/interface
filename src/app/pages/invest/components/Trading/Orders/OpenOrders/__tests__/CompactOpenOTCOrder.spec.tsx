import { columns } from 'app/pages/invest/components/Trading/Orders/OpenOrders/columns'
import * as useOpenOrderState from 'app/pages/invest/components/Trading/Orders/OpenOrders/helpers'
import { tradingQueryKeys } from 'config/queryKeys'
import React from 'react'
import { render } from 'test-utils'
import { orders } from '__fixtures__/otcOrders'
import { CompactOpenOTCOrder } from 'app/pages/invest/components/Trading/Orders/OpenOrders/CompactOpenOTCOrder'
import * as getExpiresOrderMessage from 'helpers/dates'

describe('CompactOpenOTCOrder', () => {
  const props = {
    columns: columns,
    items: orders,
    hasActions: true,
    loading: false,
    cacheQueryKey: tradingQueryKeys.getMyOpenOrdersList(
      '1234',
      '12324',
      '0x12345ef'
    )
  }
  const orderState = {
    showEmptyState: false,
    columnCount: 8,
    rowColor: jest.fn(),
    mobileRowColor: jest.fn()
  }
  it('compact open otc order matches snapshot', () => {
    jest
      .spyOn(useOpenOrderState, 'useOpenOrderState')
      .mockReturnValueOnce(orderState as any)
    jest
      .spyOn(getExpiresOrderMessage, 'getExpiresOrderMessage')
      .mockReturnValueOnce('Order expired')
    const { container } = render(<CompactOpenOTCOrder {...props} />)
    expect(container).toMatchSnapshot()
  })
})
