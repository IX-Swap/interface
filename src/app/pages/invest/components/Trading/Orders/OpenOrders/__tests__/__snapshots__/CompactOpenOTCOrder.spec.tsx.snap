// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompactOpenOTCOrder compact open otc order matches snapshot 1`] = `
<div>
  <tbody
    class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
  >
    <tr
      class="MuiTableRow-root css-8dfetm-MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-17e0zod-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-gjtr4m-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-9pwih8-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              05/10/2022
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Pair
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1b4l2pg-MuiTypography-root"
            >
              IX-APE/USD
            </h6>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Side
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            Sell
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Price
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            USD 8,500.00
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Quantity
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            1
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Total
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            USD 8,500.00
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Filled
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            0%
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-1n2mv2k"
            >
              <button
                class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root makeStyles-toggleButton-5 css-k7ezv1-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Show details
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="MuiTableRow-root css-8dfetm-MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-17e0zod-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-gjtr4m-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-9pwih8-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              05/10/2022
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Pair
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1b4l2pg-MuiTypography-root"
            >
              IX-APE/USD
            </h6>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Side
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            Sell
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Price
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            USD 8,500.00
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Quantity
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            1
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Total
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            USD 8,500.00
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Filled
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            0%
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-1n2mv2k"
            >
              <button
                class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root makeStyles-toggleButton-5 css-k7ezv1-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Show details
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="MuiTableRow-root css-8dfetm-MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-17e0zod-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-gjtr4m-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-9pwih8-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              05/10/2022
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Pair
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1b4l2pg-MuiTypography-root"
            >
              IX-APE/USD
            </h6>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Side
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            Buy
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Price
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            USD 8,500.00
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Quantity
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            5
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Total
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            USD 42,500.00
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Filled
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            20%
          </div>
        </div>
      </td>
    </tr>
    <tr
      class="MuiTableRow-root css-8dfetm-MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-17e0zod-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-gjtr4m-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-9pwih8-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              05/10/2022
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Pair
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-1b4l2pg-MuiTypography-root"
            >
              IX-APE/USD
            </h6>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Side
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            Sell
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Price
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            USD 8,500.00
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Quantity
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            1
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Total
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            USD 8,500.00
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
          >
            Filled
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-wyd1vg-MuiGrid-root"
            style="text-align: right; font-weight: 400;"
          >
            0%
          </div>
        </div>
      </td>
    </tr>
  </tbody>
</div>
`;
